### PR TITLE
Downstream Visualization improvements

### PR DESCRIPTION
**Pull request by mdekrey**
Includes the downstream branch names when getting branch details and handles nodes with partial data for when they are siblings of the loaded details.

Further overall improvements to the branch visualization.

image

### CODE REVIEW

File: ui/src/components/branch-graph/branch-graph.simulation.tsx [hunk 1]
```diff
@@ -71,38 +128,37 @@ export function useBranchSimulation<T extends BranchConfiguration>(
 	const linkingForce = useRef(
 		forceLink<WithAtom<BranchGraphNodeDatum>, WithAtom<BranchGraphLinkDatum>>(
 			[],
-		).distance((n) => Math.abs(n.source.depth - n.target.depth) * 80),
+		)
+			.distance(40)
+			.strength(0.9),
```
Comment (jeremymoritz-ps) [inline ui/src/components/branch-graph/branch-graph.simulation.tsx:133]: Nit: I'm unclear what units this number 40 represents.  Perhaps consider assigning these values to constants to make this self-documenting.

File: ui/src/components/branch-graph/branch-graph.simulation.tsx [hunk 2]
```diff
@@ -71,38 +128,37 @@ export function useBranchSimulation<T extends BranchConfiguration>(
...
 	if (simulationRef.current === undefined) {
 		simulationRef.current = forceSimulation<
 			WithAtom<BranchGraphNodeDatum>,
 			WithAtom<BranchGraphLinkDatum>
 		>([])
+			.alphaDecay(0.01)
 			.force('link', linkingForce.current)
 			.force('collide', forceCollide(6))
-			.force('spaceAround', forceManyBody().distanceMax(80))
+			.force('spaceAround', forceManyBody().distanceMax(100).strength(-100))
```
Comment (jeremymoritz-ps) [inline ui/src/components/branch-graph/branch-graph.simulation.tsx:145]: Nit: consider assigning the magic number to a variable here as well

**Review by jeremymoritz-ps — APPROVED** [review]
Nice work!

File: ui/src/components/branch-graph/branch-graph.simulation.tsx (hunk 1 above)
Comment (mdekrey) [reply ui/src/components/branch-graph/branch-graph.simulation.tsx:133]: Man, they're all pretty much magic numbers in these kinds of visualizations 😅 The best answer I have is that it sets the `forceLink` strength to 0.9, and the distance to 40. It'd be a variable like `forceLinkDistance = 40`, which I don't think is any more useful than the way the function is called. 😩

Comment (jeremymoritz-ps) [reply ui/src/components/branch-graph/branch-graph.simulation.tsx:133]: What is 40 of?  is it 40 millimeters?  40 degrees?  40 percent?

Maybe name it:

`const forceLinkDistanceInDegrees = 40;`

**Pull request merged**